cmake/Modules/Platform/Emscripten.cmake

```diff
@@ -46,10 +46,7 @@ endif()
 
 # In order for check_function_exists() detection to work, we must signal it to pass an additional flag, which causes the compilation
 # to abort if linking results in any undefined symbols. The CMake detection mechanism depends on the undefined symbol error to be raised.
-# Disable wasm in cmake checks so that (1) we do not depend on wasm support just for configuration (perhaps the user does not intend
-# to build to wasm; using asm.js only depends on js which we need anyhow), and (2) we don't have issues with a separate .wasm file
-# on the side, async startup, etc..
-set(CMAKE_REQUIRED_FLAGS "-s ERROR_ON_UNDEFINED_SYMBOLS=1 -s WASM=0")
```

Comment:
I suspect we may need `-s SINGLE_FILE=1` here, as mentioned in that comment.

Reply:
I think I can detect fastcomp here and only pass WASM=0 in that case.

Why do we need SINGLE_FILE?

Reply:
Otherwise it creates a js and a wasm file, and if it doesn't chdir to that location before running the js, the js may not find the wasm. However, that should have started to work as of recently, at least in node, so maybe that's fine.

Thinking some more, there's a downside to passing WASM=0 for fastcomp - it adds a difference between the backends, that eventually may be confusing. But, maybe that's unavoidable?

Reply:
I see..  Well if SINGLE_FILE is needed for that reason (lack of chdir()) I assume we would see that in the cmake tests we have in the test suite.

Reply:
Yeah, tests should cover the SINGLE_FILE-related issue well.

Overall, I think it's be ok to detect fastcomp here and use `-s WASM=0` for that. Not sure how easy that would be though.

Reply:
Seems like checking for fastcomp here would be possible but it would be this change more complex.

Are you sure cmake actually runs the output of the compiler while its configuring, or does it just compiler and link?   If its just compiling and linking then all this change would be is add the binaryen dependency which seems fine.

Reply:
It can run it too, yeah. Some checks, like the size of an int, it runs the program so it prints `sizeof`.

Reply:
That's a fair point. It would be a breaking change, though - people would need to add that to their CMakeLists.txt. I'm ok with that, if we document it - maybe in the Changelog and the Emscripten.cmake file (in a comment). Maybe also in the [building projects](http://kripken.github.io/emscripten-site/docs/compiling/Building-Projects.html) doc.